sklearn/datasets/_samples_generator.py

```diff
@@ -1785,6 +1785,17 @@ def _shuffle(data, random_state=None):
     return result, row_idx, col_idx
 
 
+@validate_params(
+    {
+        "shape": [tuple],
+        "n_clusters": [Interval(Integral, 1, None, closed="left")],
+        "noise": [Interval(Real, 0, None, closed="left")],
+        "minval": [Interval(Real, None, None, closed="neither")],
+        "maxval": [Interval(Real, None, None, closed="neither")],
```

Comment:
```suggestion
        "minval": [Interval(Integral, None, None, closed="neither")],
        "maxval": [Interval(Integral, None, None, closed="neither")],
```

Reply:
Right, I see no reason to restrict it to integers. You can ignore my suggestion. Can you modify the docstring instead to mark both as float ?

Reply:
Thanks for your answer, I just updated the docstring.